drivers/src/sha384acc.rs

```diff
@@ -42,13 +49,13 @@ impl Sha384Acc {
     ///
     /// * `Sha384AccOp` - On, success, an object representing the SHA384 accelerator operation.
     /// * 'None' - On failure to acquire the SHA384 Accelerator lock.
-    pub fn try_start_operation(&mut self) -> Option<Sha384AccOp> {
+    pub fn try_start_operation(&mut self, lock_state: ShaAccLockState) -> Option<Sha384AccOp> {
         let sha_acc = self.sha512_acc.regs();
 
-        if sha_acc.lock().read().lock() && sha_acc.status().read().soc_has_lock() {
-            None
+        if lock_state == ShaAccLockState::NotAcquried && sha_acc.lock().read().lock() {
+            None // SOC has the lock.
         } else {
-            // We acquired the lock, or we already have the lock (such as at startup)
```

Comment:
Didn't you mention in the meeting that in the ShaAccLockState::Acquired case you wanted to read sha_acc.lock() and ensure that it was 1, fatal error otherwise?

Reply:
My suggestion, to make all the possibilities clearer:

```rs
match lock_state {
    ShaAccLockState::Acquired => {
        if sha_acc.lock().read().lock() {
            // The uC didn't just acquire the lock, and the caller is
            // assuming the uC already has it.
            Ok(Some(Sha384AccOp {
                sha512_acc: &mut self.sha512_acc,
            }))
        } else {
            // We expected to already have the lock, but clearly we didn't (bug)
            Err(CaliptraError::DRIVER_SHA384ACC_UNEXPECTED_ACQUIRED_LOCK_STATE)
        }
    }
    ShaAccLockState::NotAcquried => {
        if sha_acc.lock().read().lock() {
            // SoC has the lock, or the uC has the lock but the caller
            // doesn't realize it (bug)
            Ok(None)
        } else {
            Ok(Some(Sha384AccOp {
                sha512_acc: &mut self.sha512_acc,
            }))
        }
    }
}
```

Reply:
I don't love how much we have to trust the caller of this function.